lib/AST/DeclContext.cpp

```diff
@@ -475,6 +475,25 @@ unsigned DeclContext::getSemanticDepth() const {
   return 1 + getParent()->getSemanticDepth();
 }
 
+bool DeclContext::mayContainMembersAccessedByDynamicLookup() const {
+  // Dynamic lookup can only find class and protocol members, or extensions of
+  // classes.
+  if (auto *NTD = dyn_cast<NominalTypeDecl>(this)) {
+    if (isa<ClassDecl>(NTD))
+      if (!isGenericContext())
+        return true;
+    if (auto *PD = dyn_cast<ProtocolDecl>(NTD))
+      if (PD->getAttrs().hasAttribute<ObjCAttr>())
+        return true;
+  } else if (auto *ED = dyn_cast<ExtensionDecl>(this)) {
+    if (auto *CD = dyn_cast<ClassDecl>(ED->getExtendedNominal()))
```

Comment:
Please use `dyn_cast_or_null` on the result of `getExtendedNominal`, since it could fail. Alternatively, `getSelfClassDecl()` would let you combine the "class" and "extension of class" cases and probably needs less code.

Reply:
Alright, I'll fix that in the other PR (this one needs a bit more debugging -- the final commit doesn't pass tests)